homeassistant/components/notify/nfandroidtv.py

```diff
@@ -169,6 +178,13 @@ def send_message(self, message="", **kwargs):
                     _LOGGER.warning("Invalid interrupt-value: %s",
                                     str(interrupt))
 
+        if icon_file is None:
+            icon_file = self._icon_file
+        payload[ATTR_FILENAME] = ('icon.png',
+                                  open(icon_file, 'rb'),
+                                  'application/octet-stream',
+                                  {'Expires': '0'})
```

Comment:
We should check if the path with `is_allowed_path` here.

Reply:
If I do that this fix doesn't work anymore. Using resources from hass_frontend seems to not be allowed.

Reply:
I'm wondering how that's possible because `is_allowed_path` has no restrictions on the location of files. At least of you are not using a `venv` I would say it's a permission issue.

Reply:
I'm developing this on a Debian VM as root within a venv. And root shouldn't have any issues with permissions. I haven't looked more deeply into it though. I just noticed, that with a default-configuration (no custom icon set) it won't work anymore.

Reply:
Since the user is allowed to specify any path to an icon, we need to use `is_allowed_path` to make sure that we don't have a malicious user send our passwords to a target. Enforce the check only when a custom path is specified.

I think the easier approach here is to remove the option to set custom icons.

Reply:
Oh I see now that setting a custom icon is added in this PR, please revert that change.

Reply:
Dynamic icons actually were a feature request a while back. And even though there are some restrcitions with the supported formats it makes sense. Just like other notifiers (like telegram) support sending images.
But I can take it out again if that's preferred.